src/amo/components/MastHead.js

```diff
@@ -34,10 +33,12 @@ export class MastHeadBase extends React.Component {
 
     return (
       <div className="MastHead">
-        <div id="tabzilla">
-          <a href="https://www.mozilla.org">Mozilla</a>
+        <div className="MastHead-top-row">
+          <div id="tabzilla">
+            <a href="https://www.mozilla.org">Mozilla</a>
+          </div>
+          <AuthenticateButton className="MastHead-auth-button" size="small" location={location} />
```

Comment:
Unfortunately need to pass `location` throughout, I didn't see a way of getting it from react-router. Could update `AuthenticateButton` to just use `window.location.pathname` instead. I'm happy with either.